### DISCUSSION TITLE
Some photos take hours to index single photo. Normal or Not?

### DISCUSSION THREAD

**Discussion by Mentolity** [photoprism/photoprism#3548] [Q&A]
I'm indexing a bunch of photos and photoprism seems to get caught up on some of the panorama photos taken on my old Galaxy S8+. Normally photos index in a few 100s of milliseconds. But these photos have taken anywhere on the order or 10s of minutes to 8 hours to fully process.

I'm running
Build [230625-17242fb07](https://docs.photoprism.app/release-notes/) in a Docker Container
On an intel NUC (i5-8265U CPU, 16GB RAM)

Restarting the container and restarting the index results in the process hanging on the same photo.

```
time="2023-07-16T18:48:58-07:00" level=info msg="convert: 74824a16739eece67a43263c09aa5227afeccacc.jpg created in 8h35m3.784618588s (convert)"
time="2023-07-16T18:49:00-07:00" level=info msg="media: created 11 thumbnails for 'PATH_TO_FILE/20180920_115129.jpg' [8h35m7.116498934s]"
```

**Comment by graciousgrey** [comment]
Do you have enough SWAP configured? High-resolution panoramic images may require [additional swap space](https://docs.photoprism.app/getting-started/troubleshooting/docker/#adding-swap) and/or physical memory above the [recommended minimum](https://docs.photoprism.app/getting-started/#system-requirements).

https://docs.photoprism.app/getting-started/troubleshooting/performance/#memory

**Reply by Mentolity** [reply]
My overall memory usage is fairly low when experiencing the issue indexing. I increased my swap size to 2GB and retried but the issue still persists.
```
root@Keep:/mnt/disk1/cache# free -mh
               total        used        free      shared  buff/cache   available
Mem:            15Gi       2.3Gi       1.0Gi        94Mi        11Gi        12Gi
Swap:          2.0Gi          0B       2.0Gi
```

Also verified that docker isn't restricted on how much ram it's allowed to use:
`docker stats`
```
CONTAINER ID   NAME               CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O        PIDS
f69dffea3307   PhotoPrism         3.23%     1.651GiB / 15.03GiB   10.98%    1.66GB / 288MB   1.1GB / 23.6GB   52
```
